### PR TITLE
release/public-v2: remove unsupported suites, fix to allow quilting with non-factors for layout

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR contains:
- remove all unsupported suites for the SRW App v1.0, keep only `FV3_GFS_v15p2` and `FV3_RRFS_v1beta`
- fix to allow quilting with non-factors for layout from @chan-hoo (cherry-picked commits 1206994e944c8b4f2b7c325f01d836efe29ff5a1 and efdef3a689c696fdb3515d789cc2b496c04256a4 from https://github.com/NOAA-EMC/fv3atm/pull/192)
- add gfortran-10 compiler flags to `ccpp/CMakeLists.txt`

No answer changes expected.

## Testing

See https://github.com/ufs-community/ufs-weather-model/pull/248.

## Dependencies

https://github.com/NOAA-EMC/fv3atm/pull/190
https://github.com/ufs-community/ufs-weather-model/pull/248

### CODE REVIEW

**Review by junwang-noaa — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks, @junwang-noaa . I think we can merge this PR now.

File: io/module_wrt_grid_comp.F90 [hunk 1]
```diff
@@ -160,6 +160,9 @@ subroutine wrt_initialize(wrt_comp, imp_state_write, exp_state_write, clock, rc)
       type(ESMF_Array)                        :: array_work, array
       type(ESMF_FieldBundle)                  :: fieldbdl_work
       type(ESMF_Field)                        :: field_work, field
+! CHJ --- Gerhard's update: quilting with non-factors for layout --- s ---
```
Comment (mkavulich) [inline io/module_wrt_grid_comp.F90]: I would suggest in-line comments like this be removed; this is an outdated and cluttered way of conveying code change information that can now be found in git commit histories.

Comment (climbfuji) [reply io/module_wrt_grid_comp.F90]: @chan-hoo made the requested changes, I cherry-picked his commit efdef3a689c696fdb3515d789cc2b496c04256a4 - please check again. Thanks!

**Review by JeffBeck-NOAA — APPROVED** [review]

**Review by JeffBeck-NOAA — APPROVED** [review]
In-line comments removed.

**Review by mkavulich — APPROVED** [review]
Looks good to me :+1:

**Comment by climbfuji** [comment]
@junwang-noaa Mike is happy with these changes, can you please merge them? Thanks.

**Comment by junwang-noaa** [comment]
merged.

**Pull request merged**